src/tests/bdd/entity/validate_non_existing_entity.feature

```diff
@@ -0,0 +1,143 @@
+      | data-source-name  | db   | 27017 | maf      | maf      | false | repo1    |  bdd-test | entities   | mongo-db | default   |
+
+    Given there exist document with id "1" in data source "data-source-name"
+    """
+    {
+        "name": "root_package",
+        "description": "",
+        "type": "dmss://system/SIMOS/Package",
+        "isRoot": true,
+        "content": [
+            {
+                "address": "$2",
+                "type": "dmss://system/SIMOS/Reference",
+                "referenceType": "link"
+            }
+        ]
+    }
+    """
+    Given there exist document with id "2" in data source "data-source-name"
+    """
+    {
+      "type": "dmss://system/SIMOS/Blueprint",
+      "name": "Employee",
+      "extends": ["dmss://system/SIMOS/NamedEntity"],
+      "attributes": [
+        {
+          "name": "isManager",
+          "attributeType": "boolean",
+          "type": "dmss://system/SIMOS/BlueprintAttribute"
+        },
+        {
+          "name": "colleagues",
+          "attributeType": "dmss://data-source-name/root_package/Employee",
+          "optional": true,
+          "dimensions": "*",
+          "contained": false,
+          "type": "dmss://system/SIMOS/BlueprintAttribute"
+        }
+      ]
+    }
+    """
+
+
+
+  Scenario: Validate non-existing entity
+    Given i access the resource url "/api/entity/validate"
+    When i make a "POST" request
+    """
+    {
+        "type": "dmss://data-source-name/root_package/Employee",
+        "name": "Ola",
+        "isManager": false
+    }
+    """
+    Then the response status should be "OK"
+    And the response should contain
+    """
+    "OK"
+    """
+
+  Scenario: Validate non-existing entity with error
+    Given i access the resource url "/api/entity/validate"
+    When i make a "POST" request
+    """
+    {
+        "type": "dmss://data-source-name/root_package/Employee",
+        "name": "Kari",
+        "age": 123
+    }
+    """
+    Then the response status should be "Bad Request"
+
+    Scenario: Validate another non-existing entity with error
+    Given i access the resource url "/api/entity/validate"
+    When i make a "POST" request
+    """
+    {
+        "type": "dmss://data-source-name/root_package/Employee",
+        "name": "Nora",
+        "salary": 123
+    }
+    """
+    Then the response status should be "Bad Request"
+
+
+    Scenario: Validate a complex non-existing entity
+    Given i access the resource url "/api/entity/validate"
+    When i make a "POST" request
+    """
+    {
+        "type": "dmss://data-source-name/root_package/Employee",
+        "name": "Legolas",
+        "isManager": false,
+        "colleagues": [
+            {
+                "address": "$5",
+                "type": "dmss://system/SIMOS/Reference",
+                "referenceType": "link"
+            }
+            ]
+    }
+    """
```

Comment:
Does there exist an extension to help us with the indentation? And do we use 2 or 4 spaces?

Reply:
from the docs https://cucumber.io/docs/gherkin/reference/ it seems like standard intendation is 2 spaces. Will update!

after a bit of research I could only find a few tools that can reformat .feature files
https://github.com/antham/ghokin
https://github.com/ducminh-phan/reformat-gherkin

it seems like these are not very popular and therefore it's probably not a good idea to include them in our pre-commit hook.